torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -1167,14 +1167,24 @@ struct to_ir {
     // and (2) only enable this OPTIONAL_NONE when loading newer
     // graphs because it is incompatible with older graphs.
     // Refinement none(name, RefinementKind::OPTIONAL_NONE);
-    if (auto optional_type = lhs_value->type()->cast<OptionalType>()) {
+    if (const auto optional_type = lhs_value->type()->cast<OptionalType>()) {
       Refinement present(name, optional_type->getElementType());
       if (tok == TK_IS) {
         return RefinementSet({}, {present});
       } else { // TK_ISNOT
         return RefinementSet({present}, {});
       }
     }
+    if (const auto union_type = lhs_value->type()->cast<UnionType>()) {
+      std::vector<TypePtr> present_types =
+          union_type->set_difference_of({NoneType::get()});
+      Refinement present(name, UnionType::create(present_types));
```

Comment:
What happens when you pass in pass in a vector of size one here ? Does it create a single element Union (which is not good). This is sort of a general question of, what should the return type of `UnionType::create` be ? Should we enforce that the caller maintains the invariant that there are multiple unique types, or should we throw if there is a single type, or return a non UnionType ?